src/Legacy/DynamoRevitVersionSelector/VersionSelectorApp.cs

```diff
@@ -74,18 +66,22 @@ public Result OnStartup(UIControlledApplication application)
 
             Products = new List<DynamoProduct>();
             int index = -1;
+            var selectedVersion = selectorData.SelectedVersion.ToString(2);
             foreach (var p in dynamoProducts)
             {
                 var path = VersionLoader.GetDynamoRevitPath(p, revitVersion);
                 if (!File.Exists(path))
                     continue;
 
-                if (path.Equals(cachedPath))
+                if (p.VersionInfo.ToString(2) == selectedVersion)
```

Comment:
Currently, we can't have 0.9.0 and 0.9.1 installed side-by-side, hence no point comparing the 3rd field.